workers/blobuploadcleanupworker/blobuploadcleanupworker.py

```diff
@@ -28,13 +28,31 @@ def __init__(self):
     def _try_cleanup_uploads(self):
         """
         Performs garbage collection on the blobupload table.
+        Will also perform garbage collection on the uploads folder in the S3 bucket,
+        if applicable.
         """
         try:
             with GlobalLock("BLOB_CLEANUP", lock_ttl=LOCK_TTL):
                 self._cleanup_uploads()
         except LockNotAcquiredException:
             logger.debug("Could not acquire global lock for blob upload cleanup worker")
-            return
+
+        if "FEATURE_CLEAN_UPLOADS_FOLDER" in app.config:
+            self._try_clean_partial_uploads()
+
+    def _try_clean_partial_uploads(self):
+        """
+        Uploads cancelled before completion leaves the possibility of untracked blobs being
+        leftover in the uploads storage folder.
+        This function cleans those blobs older than DELETION_DATE_THRESHOLD
+        """
+        for location in storage.locations:
```

Comment:
I don't think it needs to check every storage. Just the one that it's writing to, so no need to iterate over every locations.

Reply:
Due to the [_location_aware function in distributedstorage.py](https://github.com/bcaton85/quay/blob/gc-uploads/storage/distributedstorage.py#L11-L29) a location has to be passed in. Is there an easy way to get the current storage that's being written to?

Reply:
Just use the global `storage` object, which is already being imported at the top of the file.

Reply:
Resolved offline, will pass `storage.preferred_locations` instead of lopping over `storage.locations`